templates/store.php

```diff
@@ -30,7 +34,143 @@
         <div id="dokan-primary" class="dokan-single-store">
             <div id="dokan-content" class="store-page-wrap woocommerce" role="main">
 
-                <?php dokan_get_template_part( 'store-header' ); ?>
+                <?php
+                dokan_get_template_part( 'store-header' );
+
+                // Featured products section
+                if ( empty( $products_block_appearance['hide_featured_products'] ) && ( empty( $store_info['show_featured_products'] ) || 'yes' === $store_info['show_featured_products'] ) ) {
+                    $extra_product_block = 1;
+                    $featured_products   = dokan_get_featured_products( $items_to_show, $vendor_id );
+                    ?>
+                <div id="dokan-featured-products" class="dokan-products-display_section">
+                    <h2 class="products-list-heading"><?php esc_html_e( 'Featured Products', 'dokan-lite' ); ?></h2>
+
+                <?php if ( $featured_products->have_posts() ) { ?>
+                    <div class="seller-items">
+
+                        <?php woocommerce_product_loop_start(); ?>
+
+                            <?php while ( $featured_products->have_posts() ) : $featured_products->the_post(); ?>
+
+                                <?php wc_get_template_part( 'content', 'product' ); ?>
+
+                            <?php endwhile; // end of the loop. ?>
+
+                        <?php woocommerce_product_loop_end(); ?>
+
+                    </div>
+
+                <?php } else { ?>
+
+                    <p class="dokan-info"><?php esc_html_e( 'No featured products were found of this vendor!', 'dokan-lite' ); ?></p>
+
+                <?php } ?>
+                </div>
+                <?php
+                }
+
+                // Latest products section
+                if ( empty( $products_block_appearance['hide_latest_products'] ) && ( empty( $store_info['show_latest_products'] ) || 'yes' === $store_info['show_latest_products'] ) ) {
+                    $extra_product_block = 1;
+                    $latest_products     = dokan_get_latest_products( $items_to_show, $vendor_id );
+                    ?>
+                <div id="dokan-latest-products" class="dokan-products-display_section">
```

Comment:
Repeatable code block. Please introduce one function or template for reusability.